src/utilities/effects.ts

```
@@ -4182,6 +4781,39 @@ const effects = {
     }
   },
 
+  async insertDerivationGroupForPlan(derivationGroupName: string, plan: Plan | null, user: User | null): Promise<void> {
+    try {
+      if ((plan && !queryPermissions.CREATE_PLAN_DERIVATION_GROUP(user, plan)) || !plan) {
+        throwPermissionError('add a derivation group to the plan');
+      }
+
+      derivationGroupPlanLinkError.set(null);
+      if (plan !== null) {
+        const data = await reqHasura<PlanDerivationGroup>(
+          gql.CREATE_PLAN_DERIVATION_GROUP,
+          {
+            source: {
+              derivation_group_name: derivationGroupName,
+              plan_id: plan.id,
+            },
+          },
+          user,
+        );
+        const { planExternalSourceLink: sourceAssociation } = data;
+        // If the return was null, do nothing - only act on success or non-null
+        if (sourceAssociation !== null) {
+          showSuccessToast('Derivation Group Linked Successfully');
+        }
```

I think there should be some failure case for when `sourceAssociation` is null. Otherwise, this fails silently.